src/main/java/com/google/devtools/build/lib/rules/python/PyCommon.java

```diff
@@ -158,16 +165,39 @@ public void addCommonTransitiveInfoProviders(RuleConfiguredTargetBuilder builder
    *     addSkylarkTransitiveInfo(PYTHON_SKYLARK_PROVIDER_NAME, createSourceProvider(...))
    */
   public static SkylarkClassObject createSourceProvider(
-      NestedSet<Artifact> transitivePythonSources, boolean isUsingSharedLibrary) {
+      NestedSet<Artifact> transitivePythonSources,
+      boolean isUsingSharedLibrary,
+      NestedSet<PathFragment> imports) {
     return NativeClassObjectConstructor.STRUCT.create(
         ImmutableMap.<String, Object>of(
             TRANSITIVE_PYTHON_SRCS,
             SkylarkNestedSet.of(Artifact.class, transitivePythonSources),
             IS_USING_SHARED_LIBRARY,
-            isUsingSharedLibrary),
+            isUsingSharedLibrary,
+            IMPORTS,
+            SkylarkNestedSet.of(String.class,
+                                makeImportStrings(imports ))
+            ),
         "No such attribute '%s'");
   }
 
+  /**
+   * Converts {@link NestedSet} of {@link PathFragment} to
+   * {@link NestedSet} of {@link String}
+   * @param imports - {@link NestedSet} of {@link PathFragment}
+   * @return - {@link NestedSet} of {@link String}
+   */
+  public static NestedSet<String> makeImportStrings(
+      NestedSet<PathFragment> imports) {
+    final NestedSetBuilder<String> builder = NestedSetBuilder.compileOrder();
+
+    for (PathFragment pathFragment : imports) {
+      builder.add(pathFragment.toString());
```

Comment:
Sorry for my late reply. I will add it asap.

Reply:
I am sorry for not understanding what you are trying to suggest here. I think (I may be completely wrong) that you are suggesting the proposed change? @dslomov can you comment?

Reply:
Friendly ping @dslomov :)

Reply:
Please read https://docs.bazel.build/versions/master/skylark/depsets.html.

What you are doing here is flattening the depset of path fragments to convert them to strings.
Since this is happening for every python rule, that will bloat the memory.
There are two approaches you can take instead:
1) Use `NestedSet<String>` instead of `NestedSet<PathFragment>` everywhere.
2) Make a value of 'imports' an `Iterable` that lazily flattens a nested set on iteration.

My preferred approach is (1) but I do not know how expensive it is to keep Strings instead of PathFragments.
(2) is ok as well.